scripts/bootstrap.sh.in

```diff
@@ -41,25 +52,66 @@ stop_salt_minion_service() {
         ${SYSTEMCTL} stop salt-minion.service 2>/dev/null || true
 }
 
+configure_yum_repositories() {
+    local -r repositories_pillar_path=/srv/scality/metalk8s-@VERSION@/pillar/repositories.sls
+    local online_mode
+
+    online_mode=$(awk '$1 == "online_mode:" { print $2 }' "$repositories_pillar_path")
+
+    if [[ ${online_mode,,} = true ]]; then
+        configure_salt_repository
+    else
+        configure_yum_local_repositories
+    fi
+
+    "$YUM" clean all
+}
+
 configure_salt_repository() {
-        ${RPM} --import https://repo.saltstack.com/yum/redhat/7/x86_64/archive/2018.3.3/SALTSTACK-GPG-KEY.pub
-        cat > /etc/yum.repos.d/saltstack.repo << EOF
-[saltstack-repo]
+    "$RPM" --import https://repo.saltstack.com/yum/redhat/7/x86_64/archive/2018.3.3/SALTSTACK-GPG-KEY.pub
+    cat > /etc/yum.repos.d/saltstack.repo << EOF
+[saltstack]
 name=SaltStack repo for RHEL/CentOS \$releasever
 baseurl=https://repo.saltstack.com/yum/redhat/\$releasever/\$basearch/archive/2018.3.3
 enabled=1
 gpgcheck=1
 gpgkey=https://repo.saltstack.com/yum/redhat/\$releasever/\$basearch/archive/2018.3.3/SALTSTACK-GPG-KEY.pub
 EOF
-        ${YUM} clean expire-cache
+}
+
+configure_yum_local_repositories() {
+    for repository in "${!GPGCHECK_REPOSITORIES[@]}"; do
+        configure_yum_local_repository "$repository" \
+            "${GPGCHECK_REPOSITORIES[$repository]}"
+    done
+}
+
+configure_yum_local_repository() {
+    local -r repo_name=$1 gpgcheck=${2:-0}
+    local -r repo_path=/srv/scality/metalk8s-@VERSION@/packages/$repo_name-el7
+    local gpg_keys
+
+    gpg_keys=$(
+        find "$repo_path" -maxdepth 1 -name "RPM-GPG-KEY-*" \
+	    -printf "file://%p\n"
+    )
+
+    cat > /etc/yum.repos.d/"$repo_name".repo << EOF
+[$repo_name]
+name=$repo_name
+baseurl=file://$repo_path
+enabled=1
+gpgcheck=$gpgcheck
+${gpg_keys:+gpgkey=$gpg_keys}
+EOF
```

Comment:
```
baseurl=http://127.0.0.1:8080/metalk8s-kubernetes-el$releasever
repo_gpg_check=0
file=///srv/scality/metalk8s-2.0/packages/metalk8s-kubernetes-el7/RPM-GPG-KEY-metalk8s-kubernetes-7_2
gpgcheck=1
```
Is this expected ? the file=

Reply:
No, it's because salt is stupid and doesn't know how to manage correctly repositories configurations
In the yum good practices, gpg files must be separated by a new line, I will change this with a space.